https://invent.kde.org/pim/kleopatra/-/commit/a3684049dd4f0d2d47dcc6605ff31ce753254ef9
https://bugs.gentoo.org/865903
https://bugs.kde.org/show_bug.cgi?id=458044

From: =?UTF-8?q?Ingo=20Kl=C3=B6cker?= <dev@ingo-kloecker.de>
Date: Fri, 19 Aug 2022 13:43:41 +0200
Subject: [PATCH] Use ReceiveKeysJob for refreshing OpenPGP keys

This refreshes the OpenPGP key from the configured key server.

GnuPG-bug-id: 5903

CHANGELOG: Fix building against QGpgME 1.18.0
BUG: 458044
FIXED-IN: 22.08.1

(cherry picked from commit 87d8b00d4b2286489d5fadc9cfa07f1d721cdfe3)
--- a/src/commands/refreshcertificatecommand.cpp
+++ b/src/commands/refreshcertificatecommand.cpp
@@ -19,7 +19,7 @@
 #include <QGpgME/Protocol>
 #ifdef QGPGME_SUPPORTS_KEY_REFRESH
 #include <QGpgME/RefreshKeysJob>
-#include <QGpgME/RefreshOpenPGPKeysJob>
+#include <QGpgME/ReceiveKeysJob>
 #endif
 
 #include <gpgme++/importresult.h>
@@ -44,7 +44,7 @@ public:
     void cancel();
 
 #ifdef QGPGME_SUPPORTS_KEY_REFRESH
-    std::unique_ptr<QGpgME::RefreshOpenPGPKeysJob> startOpenPGPJob();
+    std::unique_ptr<QGpgME::ReceiveKeysJob> startOpenPGPJob();
     std::unique_ptr<QGpgME::RefreshKeysJob> startSMIMEJob();
 #endif
     void onOpenPGPJobResult(const ImportResult &result);
@@ -136,19 +136,19 @@ void RefreshCertificateCommand::Private::cancel()
 }
 
 #ifdef QGPGME_SUPPORTS_KEY_REFRESH
-std::unique_ptr<QGpgME::RefreshOpenPGPKeysJob> RefreshCertificateCommand::Private::startOpenPGPJob()
+std::unique_ptr<QGpgME::ReceiveKeysJob> RefreshCertificateCommand::Private::startOpenPGPJob()
 {
-    std::unique_ptr<QGpgME::RefreshOpenPGPKeysJob> refreshJob{QGpgME::openpgp()->refreshOpenPGPKeysJob()};
+    std::unique_ptr<QGpgME::ReceiveKeysJob> refreshJob{QGpgME::openpgp()->receiveKeysJob()};
     Q_ASSERT(refreshJob);
 
-    connect(refreshJob.get(), &QGpgME::RefreshOpenPGPKeysJob::result,
+    connect(refreshJob.get(), &QGpgME::ReceiveKeysJob::result,
             q, [this](const GpgME::ImportResult &result) {
                 onOpenPGPJobResult(result);
             });
     connect(refreshJob.get(), &QGpgME::Job::progress,
             q, &Command::progress);
 
-    const GpgME::Error err = refreshJob->start({key});
+    const GpgME::Error err = refreshJob->start({QString::fromLatin1(key.primaryFingerprint())});
     if (err) {
         showError(err);
         return {};
GitLab
